src/gadgets/wnn.rs

```diff
@@ -141,37 +144,34 @@ impl<F: PrimeFieldBits> WnnInstructions<F> for WnnChip<F> {
         );
 
         // Assign the inputs to the first column of the hash chip
-        let mut permuted_inputs = vec![];
-        layouter.assign_region(|| "permuted inputs", |mut region| {
-            for (i, input) in inputs.iter().enumerate() {
-                let input_cell = region.assign_advice(
-                    || format!("input {}", i),
-                    self.config.bit2num_chip_config.input,
-                    i,
-                    || Value::known(F::from(*input as u64)),
-                )?;
-
-                permuted_inputs.push(input_cell);
-            }
-            Ok(())
-        })?;
+
+        let permuted_inputs = layouter.assign_region(
+            || "permuted inputs",
+            |mut region| {
+                let mut permuted_inputs = vec![];
+                for (i, input) in inputs.iter().enumerate() {
+                    let input_cell = region.assign_advice(
+                        || format!("input {}", i),
+                        self.config.bit2num_chip_config.input,
+                        i,
+                        || Value::known(F::from(*input as u64)),
+                    )?;
+
+                    permuted_inputs.push(input_cell);
+                }
+                Ok(permuted_inputs)
+            },
+        )?;
 
         // Convert the input bits to a group of field element that can be hashed
-        let joint_inputs : Vec<AssignedCell<F, F>> = permuted_inputs
+        let joint_inputs = permuted_inputs
             .chunks_exact(self.config.bit2num_chip_config.bit2num_config.num_bit_size)
-            .map(|chunk| {
-                bit2num_chip
-                    .convert_le(
-                        &mut layouter,
-                        Vec::from(chunk),
-                    )
-                    .unwrap()
-            })
-            .collect();
+            .map(|chunk| bit2num_chip.convert_le(&mut layouter, Vec::from(chunk)))
+            .collect::<Result<Vec<_>, _>>()?;
```

Comment:
I got rid of the `unwrap()` call. Apparently, `collect()` is smart enough to turn a `Vec<Result<_, _>>` into a `Result<Vec<_>, _>` if you provide the type information. This lets us use the `?` operator at the end which makes the final result `Vec<AssignedCell<F, F>>`, as before.